csrc/multidevice/utils.h

```diff
@@ -86,8 +86,22 @@ bool haveDifferentShardings(
 // Returns whether a resharding expr reshards an inner axis
 bool isInnerResharding(Expr* expr);
 
+// Shards all tensors in tvs like reference
 void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs);
 
+// Shards all TVs from the output of from expressions to the outputs of the to
+// expressions like the reference TV.
+void shardBetween(
+    const std::vector<Expr*>& from,
+    const std::vector<Expr*>& to,
+    TensorView* ref);
+
+// Shards all TVs between from_tvs and to_tvs like the reference TV.
+void shardBetween(
+    const std::vector<TensorView*>& from_tvs,
+    const std::vector<TensorView*>& to_tvs,
```

Comment:
```suggestion
    const std::vector<TensorView*>& from,
    const std::vector<TensorView*>& to,
```
The type is already clear that these arguments must be TVs.